pkg/crc/cluster/pullsecret.go

```diff
@@ -0,0 +1,153 @@
+package cluster
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	cmdConfig "github.com/code-ready/crc/cmd/crc/cmd/config"
+	crcConfig "github.com/code-ready/crc/pkg/crc/config"
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/crc/logging"
+	"github.com/code-ready/crc/pkg/crc/validation"
+	crcversion "github.com/code-ready/crc/pkg/crc/version"
+	"github.com/zalando/go-keyring"
+	terminal "golang.org/x/term"
+	"gopkg.in/AlecAivazis/survey.v1"
+)
+
+const (
+	keyringService = "crc"
+	keyringUser    = "pull-secret"
+)
+
+type PullSecretLoader interface {
+	Value() (string, error)
+}
+
+type interactivePullSecretLoader struct {
+	nonInteractivePullSecretLoader *nonInteractivePullSecretLoader
+}
+
+func NewInteractivePullSecretLoader(config crcConfig.Storage) PullSecretLoader {
+	return &PullSecretMemoizer{
+		Getter: &interactivePullSecretLoader{
+			nonInteractivePullSecretLoader: &nonInteractivePullSecretLoader{
+				config: config,
+			},
+		},
+	}
+}
+
+func (loader *interactivePullSecretLoader) Value() (string, error) {
```

Comment:
The loader is what retrieves the pullsecret from a keyring?

Isn't Retriever or Reader a better fit? A loader sounds more like a skidloeader, or a helper program to get your software up and going, like those for the C64... a tape loader?

Also wondering what the distinction between Interactive and Non-interactive is.

Reply:
More about what triggers the non-interactive flow and when the interactive. as you said, the daemon would not ask... not in that case it would have to error and allow the resolution to happen from where? the daemon is only used by the systray, so in that case this has to be provided beforehand anyways, but this does not support an error feedback for this to 'prompt' for this by a dialog.

Reply:
Getter is a very Java-like term, but perhaps it is more about lead-up to this.... StorageHandler... Store... KeyringLoader? Ot misses perhaps the target to indicate how it is loaded; reason why I mention it is generic. Perhaps purposely?